force-app/main/default/classes/DMLOperations.cls

```diff
@@ -145,66 +154,142 @@ public with sharing class DMLOperations {
      * @param oppNames The list of Opportunity names.
      */
     public static void upsertOpportunities(String accountName, List<String> oppNames) {
-        // Loop through each Opportunity name
-            // Create an Opportunity with the given name
-            // Add the Opportunity to the list
-    }
+        // Query for the account
+        List<Account> accList = [SELECT Id, Name FROM Account WHERE Name = :accountName LIMIT 1];
+        Account accProvided;
+        
+        if (accList.isEmpty()) {
+            // No account found, so create a new one
+            accProvided = new Account(Name = accountName);
+            insert accProvided;
+        } else {
+            // Use the existing account
+            accProvided = accList[0];
+        }
+    
+        // Initialize the map to store existing Opportunities
+        Map<String, Opportunity> exOppos = new Map<String, Opportunity>();
```

Comment:
Here, we can be more explicit with our variable name: `existingOppsMap`. The use of **ex** has too many connotations that reference the idea of being _former_ (eg, ex-wife). So, at first glance, someone reading your code might think '_what's an ex-opportunity_?' Also, we tend to use `opps` for plural Opportunities as opposed to `oppos`. Finally, it's a good idea to include the word `Map` in your map variable name. It immediately let's us know what we're dealing with.